### ISSUE TITLE
Error when using img_ tag

### ISSUE THREAD

**Issue by tysonzero** [liqula/react-hs#72]
uncaught exception in Haskell main thread: Invariant Violation: img is a void element tag and must neither have `children` nor use `dangerouslySetInnerHTML`. Check the render method of null.

I cannot use `img_` at all currently, here is a minimal failing example:

```
{-# LANGUAGE DataKinds, OverloadedStrings #-}

module Main (main) where

import React.Flux (View, img_, mkView, reactRenderView)

main :: IO ()
main = reactRenderView "content" baseView

baseView :: View '[]
baseView = mkView "" $ img_ [] ""
```

**Comment by tysonzero** [comment]
It seems like we need two separate primitives for defining elements, `term'` as we have been using so far for elements that can have children, and `term'_` for elements like `img_` that cannot. Would also avoid the annoying empty string at the end of `img_`.

**Comment by fisx** [comment]
Thanks for reporting this.

Related to #65.

I still can't promise anything, but I hope I'll have more time for open issues and PRs this week.

**Comment by tysonzero** [comment]
Alright thank you!

Are you planning on adding any new core developers anytime soon, it seems like you are generally quite busy.

**Comment by fisx** [comment]
related: https://hackage.haskell.org/package/type-of-html

this is an approach to bring correct html and type-correct html closer together.  i would love to have time to try and work this into React.Flux.DOM.

**Comment by fisx** [comment]
has this been fixed in #65?

**Comment by tysonzero** [comment]
@fisx Yes it has
